server/handlers/project_command_output_handler.go

```diff
@@ -79,21 +127,34 @@ func (p *DefaultProjectCommandOutputHandler) Handle() {
 	}
 }
 
-func (p *DefaultProjectCommandOutputHandler) Clear(ctx models.ProjectCommandContext) {
-	p.ProjectCmdOutput <- &models.ProjectCmdOutputLine{
+func (p *AsyncProjectCommandOutputHandler) Clear(ctx models.ProjectCommandContext) {
+	if !p.featureEnabled(ctx) {
+		return
+	}
```

Comment:
dude i see the opportunity for another delegate here 😉 

```go
type FeatureAwareOutputHandler {
  ProjectCommandOutputHandler
}

func (p *FeatureAwareOutputHandler) Clear(ctx models.ProjectCommandContext) {
    if !p.featureEnabled(ctx) {
        return
    }
    return ProjectCommandOutputHandler.Clear(ctx)
}
...
```

That way when we're done with the feature (after it's launched) We just have to delete this proxy.

Reply:
Yeah, I wanted to avoid creating a decorator :D But it makes sense in this scenario